compiler/vm/vm_enums.nim

```diff
@@ -32,6 +41,8 @@ type
     opcLdObj,  # a = b.c
     opcLdObjAddr, # a = addr(b.c)
     opcWrObj,  # a.b = c
+    # TODO: needs a better name
```

Comment:
How about `opcWrObjFld`?

Reply:
I should have put a newline there; the to-do is targeted at `opcLdRef`.

`opcLdRef` loads a handle to the location which is referenced/targeted by the `ref` value in register 'b' into register 'a'

Reply:
So it's creating a new reference/handle to the same value/location.

I'm thinking out loud below:

From a very low level generic register machine perspective we're:
- copying a value from B to A
- updating/storing a value to track the new ref

From the computational model perspective we're creating a new handle to a location and tracking the new handle for a various reasons one of which is knowing if a location is orphaned or not.

At present many of the operations, at least in terms of vocabulary use generic VM terms. That terminology hasn't been updated and it might not need to. At least at this juncture I'm not sure which to use, because `opcCloneHdl` feels like the most natural thing but it's at odds with the other stuff.

Helpful line of inquiry?

Reply:
Thinking out loud:

A handle is synonymous with naming a chunk of memory (location) and assuming its schema (type). Making a handle for a location and shoving it into a register is equivalent to putting an address in a register.

I reread the operation and it seems like that's what's happen, but I'm on the phone and I couldn't trace through all the relevant code.
---

I noticed a potential difference in our conceptual models, I consider a location a register... I'm less certain this is a good idea. Perhaps that's why I'm perceiving move/copy but you're not as I don't think you consider the register a location. Moreover, I think the perspective I was taking is not entirely correct. Although, it does have partial utility in that we can describe the fact that pointer mutations don't have to be visible to registers because in my model registers hold values.